isis/tests/FunctionalTestsKaguyatc2isis.cpp

```diff
@@ -0,0 +1,88 @@
+#include <QTemporaryDir>
+
+#include "kaguyatc2isis.h"
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+
+#include "gtest/gtest.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/kaguyatc2isis.xml").expanded();
+
+TEST(kaguyatc2isisTest, kaguyatc2isisTestDefault) {
+  Pvl appLog;
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/kaguyatc2isisTEMP.cub";
+  QVector<QString> args = {"from=data/kaguyatc2isis/TC1S2B0_01_05186N225E0040_mini.lbl", "to="+cubeFileName};
+
+  UserInterface options(APP_XML, args);
+  try {
+    kaguyatc2isis(options, &appLog);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to ingest Kaguya TC image: " <<e.toString().toStdString().c_str() << std::endl;
+  }
+
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+  // Dimensions group
+  PvlGroup &dims = isisLabel->findGroup("Dimensions", Pvl::Traverse);
+  ASSERT_EQ(int(dims["Samples"]), 3208);
+  ASSERT_EQ(int(dims["Lines"]), 3);
+  ASSERT_EQ(int(dims["Bands"]), 1);
```

Comment:
Use Cube::lineCount, sampleCount, and bandCount instead of directly accessing the label.